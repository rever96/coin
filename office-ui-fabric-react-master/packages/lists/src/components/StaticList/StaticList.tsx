import * as React from 'react';
import { IStaticListProps } from './StaticList.types';

function defaultRowRenderer<T>(item: T, index: number): JSX.Element {
  return <li key={index}>{item}</li>;
}

class StaticList<T> extends React.Component<IStaticListProps<T>> {
  public render(): JSX.Element {
    const { items = [], as: RootTag = 'ul', className, children = defaultRowRenderer } = this.props;

    return <RootTag className={className}>{items.map(children)}</RootTag>;
  }
}

export { StaticList };
