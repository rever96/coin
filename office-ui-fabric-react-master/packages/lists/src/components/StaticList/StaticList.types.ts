interface IStaticListProps<T> {
  as?: keyof JSX.IntrinsicElements;
  items?: ReadonlyArray<T>;
  children?: (item: T, index: number) => React.ReactNode;
  className?: string;
}

export { IStaticListProps };
