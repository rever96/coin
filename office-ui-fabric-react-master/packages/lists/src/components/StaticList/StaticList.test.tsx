import * as React from 'react';
import { StaticList } from './StaticList';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

function generateItems(count: number): number[] {
  const itemsArray: number[] = [];
  for (let i = 0; i < count; i++) {
    itemsArray.push(i);
  }
  return itemsArray;
}

const COUNT: number = 50;
let items: ReadonlyArray<number>;

describe('StaticList', () => {
  beforeEach(() => {
    items = generateItems(COUNT);
  });

  it('renders empty list correctly', () => {
    const mockRenderFunction = jest.fn();
    const component = renderer.create(<StaticList>{mockRenderFunction}</StaticList>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders empty list with custom root tag correctly', () => {
    const mockRenderFunction = jest.fn();
    const component = renderer.create(<StaticList as={'ol'}>{mockRenderFunction}</StaticList>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders default unordered-list correctly', () => {
    const component = renderer.create(<StaticList items={items} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders unordered-list with child render function correctly', () => {
    const mockRenderFunction = (_item: number, index: number) => <li key={index}>{`Item #${index}`}</li>;
    const component = renderer.create(<StaticList items={items}>{mockRenderFunction}</StaticList>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders ordered-list correctly', () => {
    const mockRenderFunction = (_item: number, index: number) => <li key={index}>{`Item #${index}`}</li>;
    const component = renderer.create(
      <StaticList items={items} as={'ol'}>
        {mockRenderFunction}
      </StaticList>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders no children if no items provided', () => {
    const mockRenderFunction = jest.fn();
    const wrapper = shallow(<StaticList>{mockRenderFunction}</StaticList>);

    expect(mockRenderFunction).not.toHaveBeenCalled();
    expect(wrapper.children().length).toBe(0);
  });

  it('invokes render function per item provided', () => {
    const mockRenderFunction = jest.fn();

    shallow(<StaticList items={items}>{mockRenderFunction}</StaticList>);

    expect(mockRenderFunction).toHaveBeenCalledTimes(COUNT);
  });

  it('renders custom root tag if provided', () => {
    const mockRenderFunction = jest.fn();

    const wrapper = shallow(
      <StaticList as={'ol'} items={items}>
        {mockRenderFunction}
      </StaticList>
    );

    expect(wrapper.matchesElement(<ol />)).toBe(true);
  });

  it('renders child render function per item provided', () => {
    const wrapper = shallow(
      <StaticList items={items}>{(_item: number, index: number) => <li key={index}>{`Item #${index}`}</li>}</StaticList>
    );

    expect(wrapper.children().length).toBe(COUNT);
  });

  it('re-renders when items array mutates', () => {
    const wrapper = shallow(
      <StaticList items={items}>{(_item: number, index: number) => <li key={index}>{`Item #${index}`}</li>}</StaticList>
    );

    wrapper.setProps<'items'>({ items: [...items, ...items] });

    expect(wrapper.children().length).toBe(100);
  });

  it('re-renders when "as" value mutates', () => {
    const mockRenderFunction = jest.fn();

    const wrapper = shallow(<StaticList items={items}>{mockRenderFunction}</StaticList>);

    expect(wrapper.matchesElement(<ul />)).toBe(true);

    wrapper.setProps<'as'>({ as: 'ol' });

    expect(wrapper.matchesElement(<ol />)).toBe(true);
  });

  it('re-renders when children mutate', () => {
    const wrapper = shallow(
      <StaticList items={items}>{(_item: number, index: number) => <li className="foo" key={index}>{`Item #${index}`}</li>}</StaticList>
    );

    expect(wrapper.find('.foo').length).toBe(COUNT);

    wrapper.setProps({ children: (_item: number, index: number) => <li className="bar" key={index}>{`Item #${index}`}</li> });

    expect(wrapper.find('.bar').length).toBe(COUNT);
  });
});
