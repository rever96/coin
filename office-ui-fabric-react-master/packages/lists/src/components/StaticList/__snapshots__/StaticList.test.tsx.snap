// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticList renders default unordered-list correctly 1`] = `
<ul>
  <li>
    0
  </li>
  <li>
    1
  </li>
  <li>
    2
  </li>
  <li>
    3
  </li>
  <li>
    4
  </li>
  <li>
    5
  </li>
  <li>
    6
  </li>
  <li>
    7
  </li>
  <li>
    8
  </li>
  <li>
    9
  </li>
  <li>
    10
  </li>
  <li>
    11
  </li>
  <li>
    12
  </li>
  <li>
    13
  </li>
  <li>
    14
  </li>
  <li>
    15
  </li>
  <li>
    16
  </li>
  <li>
    17
  </li>
  <li>
    18
  </li>
  <li>
    19
  </li>
  <li>
    20
  </li>
  <li>
    21
  </li>
  <li>
    22
  </li>
  <li>
    23
  </li>
  <li>
    24
  </li>
  <li>
    25
  </li>
  <li>
    26
  </li>
  <li>
    27
  </li>
  <li>
    28
  </li>
  <li>
    29
  </li>
  <li>
    30
  </li>
  <li>
    31
  </li>
  <li>
    32
  </li>
  <li>
    33
  </li>
  <li>
    34
  </li>
  <li>
    35
  </li>
  <li>
    36
  </li>
  <li>
    37
  </li>
  <li>
    38
  </li>
  <li>
    39
  </li>
  <li>
    40
  </li>
  <li>
    41
  </li>
  <li>
    42
  </li>
  <li>
    43
  </li>
  <li>
    44
  </li>
  <li>
    45
  </li>
  <li>
    46
  </li>
  <li>
    47
  </li>
  <li>
    48
  </li>
  <li>
    49
  </li>
</ul>
`;

exports[`StaticList renders empty list correctly 1`] = `<ul />`;

exports[`StaticList renders empty list with custom root tag correctly 1`] = `<ol />`;

exports[`StaticList renders ordered-list correctly 1`] = `
<ol>
  <li>
    Item #0
  </li>
  <li>
    Item #1
  </li>
  <li>
    Item #2
  </li>
  <li>
    Item #3
  </li>
  <li>
    Item #4
  </li>
  <li>
    Item #5
  </li>
  <li>
    Item #6
  </li>
  <li>
    Item #7
  </li>
  <li>
    Item #8
  </li>
  <li>
    Item #9
  </li>
  <li>
    Item #10
  </li>
  <li>
    Item #11
  </li>
  <li>
    Item #12
  </li>
  <li>
    Item #13
  </li>
  <li>
    Item #14
  </li>
  <li>
    Item #15
  </li>
  <li>
    Item #16
  </li>
  <li>
    Item #17
  </li>
  <li>
    Item #18
  </li>
  <li>
    Item #19
  </li>
  <li>
    Item #20
  </li>
  <li>
    Item #21
  </li>
  <li>
    Item #22
  </li>
  <li>
    Item #23
  </li>
  <li>
    Item #24
  </li>
  <li>
    Item #25
  </li>
  <li>
    Item #26
  </li>
  <li>
    Item #27
  </li>
  <li>
    Item #28
  </li>
  <li>
    Item #29
  </li>
  <li>
    Item #30
  </li>
  <li>
    Item #31
  </li>
  <li>
    Item #32
  </li>
  <li>
    Item #33
  </li>
  <li>
    Item #34
  </li>
  <li>
    Item #35
  </li>
  <li>
    Item #36
  </li>
  <li>
    Item #37
  </li>
  <li>
    Item #38
  </li>
  <li>
    Item #39
  </li>
  <li>
    Item #40
  </li>
  <li>
    Item #41
  </li>
  <li>
    Item #42
  </li>
  <li>
    Item #43
  </li>
  <li>
    Item #44
  </li>
  <li>
    Item #45
  </li>
  <li>
    Item #46
  </li>
  <li>
    Item #47
  </li>
  <li>
    Item #48
  </li>
  <li>
    Item #49
  </li>
</ol>
`;

exports[`StaticList renders unordered-list with child render function correctly 1`] = `
<ul>
  <li>
    Item #0
  </li>
  <li>
    Item #1
  </li>
  <li>
    Item #2
  </li>
  <li>
    Item #3
  </li>
  <li>
    Item #4
  </li>
  <li>
    Item #5
  </li>
  <li>
    Item #6
  </li>
  <li>
    Item #7
  </li>
  <li>
    Item #8
  </li>
  <li>
    Item #9
  </li>
  <li>
    Item #10
  </li>
  <li>
    Item #11
  </li>
  <li>
    Item #12
  </li>
  <li>
    Item #13
  </li>
  <li>
    Item #14
  </li>
  <li>
    Item #15
  </li>
  <li>
    Item #16
  </li>
  <li>
    Item #17
  </li>
  <li>
    Item #18
  </li>
  <li>
    Item #19
  </li>
  <li>
    Item #20
  </li>
  <li>
    Item #21
  </li>
  <li>
    Item #22
  </li>
  <li>
    Item #23
  </li>
  <li>
    Item #24
  </li>
  <li>
    Item #25
  </li>
  <li>
    Item #26
  </li>
  <li>
    Item #27
  </li>
  <li>
    Item #28
  </li>
  <li>
    Item #29
  </li>
  <li>
    Item #30
  </li>
  <li>
    Item #31
  </li>
  <li>
    Item #32
  </li>
  <li>
    Item #33
  </li>
  <li>
    Item #34
  </li>
  <li>
    Item #35
  </li>
  <li>
    Item #36
  </li>
  <li>
    Item #37
  </li>
  <li>
    Item #38
  </li>
  <li>
    Item #39
  </li>
  <li>
    Item #40
  </li>
  <li>
    Item #41
  </li>
  <li>
    Item #42
  </li>
  <li>
    Item #43
  </li>
  <li>
    Item #44
  </li>
  <li>
    Item #45
  </li>
  <li>
    Item #46
  </li>
  <li>
    Item #47
  </li>
  <li>
    Item #48
  </li>
  <li>
    Item #49
  </li>
</ul>
`;
